### PR TITLE
Add aws assume role support

### PR DESCRIPTION
**Pull request by RamanaReddy0M**
- closes: #550 

```console
- # provider is the name of the provider
  provider: aws
  # id is the name of the provider id
  id: staging
  # aws_access_key is the access key for AWS account
  aws_access_key: AKIAXXXXXXXXXXXXXX
  # aws_secret_key is the secret key for AWS account
  aws_secret_key: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
  # aws_session_token session token for temporary security credentials retrieved via STS (optional)
  aws_session_token: xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
  # assume_role_name is the name of the role to assume (optional)
  assume_role_name: xxxxxxxx
  # account_ids is the aws account ids which has similar assumed role name (optional)
  account_ids:
    - xxxxxxxxxxx
    - xxxxxxxxxxx
```

### CODE REVIEW

File: pkg/providers/aws/aws.go [hunk 1]
```diff
@@ -20,13 +21,69 @@ import (
...
+func (p *ProviderOptions) ParseOptionBlock(block schema.OptionBlock) error {
+	accessKey, ok := block.GetMetadata(apiAccessKey)
+	if !ok {
+		return &schema.ErrNoSuchKey{Name: apiAccessKey}
+	}
+	accessToken, ok := block.GetMetadata(apiSecretKey)
+	if !ok {
+		return &schema.ErrNoSuchKey{Name: apiSecretKey}
+	}
+	p.Token, _ = block.GetMetadata(sessionToken)
```
Comment (dogancanbakir) [inline pkg/providers/aws/aws.go:48]: are these optional?

Comment (RamanaReddy0M) [reply pkg/providers/aws/aws.go:48]: yes

**Review by dogancanbakir — APPROVED** [review]

**Review by ehsandeep — COMMENTED** [review]
@RamanaReddy0M Can we update the docs at https://docs.projectdiscovery.io/tools/cloudlist/providers#aws-amazon-web-services as well?

**Review by ehsandeep — CHANGES_REQUESTED** [review]
```console
./cloudlist -pc pc.yaml 

  _______             _____     __ 
 / ___/ /__  __ _____/ / (_)__ / /_
/ /__/ / _ \/ // / _  / / (_-</ __/
\___/_/\___/\_,_/\_,_/_/_/___/\__/ 

		projectdiscovery.io

[INF] Current cloudlist version 1.0.8 (latest)
[FTL] Could not create runner: yaml: unmarshal errors:
  line 7: cannot unmarshal !!seq into string
```

config parsing error on default run, works fine with provider filter.

**Review by ehsandeep — APPROVED** [review]

**Pull request merged**